test/jdk/java/nio/channels/DatagramChannel/Loopback.java

```diff
@@ -163,12 +163,16 @@ static void test(ProtocolFamily family, InetAddress group, NetworkInterface ni)
                         if (src.mismatch(dst) != -1) {
                             System.out.println("src: " + src + "not equal to dst: " + dst);
                             dst.clear();
+                            // We're not supposed to receive stray datagrams here,
+                            // so respin with 3 full seconds as this shouldn't happen too often
+                            Thread.sleep(3000);
```

Comment:
Sorry for being unclear - please remove that Thread.sleep().

What I meant is that the `continue` statement below will lead us to calling `sel.select(3000);` again at line 155, and since we're not expecting to receive any datagram, it will block for 3 seconds again. 

Anyway - that's what the test was doing already if receiving a datagram from a socket where `senderPort != localPort` - so that's probably OK to leave it like that.

Please revert the last change - what you had before was good.

Reply:
Sorry foe the mis-understand, Thread.sleep and InterruptedException has been deleted.